docs/hardware_keyboard_guidelines.md

```diff
@@ -1,18 +1,94 @@
 # QMK Keyboard Guidelines
 
-We welcome all keyboard projects into QMK, but ask that you try to stick to a couple guidelines that help us keep things organised and consistent.
+Since starting QMK has grown by leaps and bounds thanks to people like you who contributed to creating and maintaining our community keyboards. As we have grown we have discovered some patterns that work well, and ask that you conform to them to make it easier for other people to make use of your hard work.
 
 ## Naming Your Keyboard/Project
 
-All names should be lowercase alphanumeric, and separated by an underscore (`_`), but not begin with one. Your directory and your `.h` and `.c` files should have exactly the same name. All folders should follow the same format. `test`, `keyboard`, and `all` are reserved by make and are not a valid name for a keyboard.
+All keyboard names are in lower case, consisting only of letters, numbers, and underscore (`_`). Names may not begin with an underscore. Forward slash (`/`) is used as a sub-folder separation character.
```

Comment:
```suggestion
All keyboard names are in lower case, consisting only of letters, numbers and underscores (`_`). Names may not begin with an underscore. Forward slash (`/`) is used as a subdirectory separation character.
```
Use more generic / platform-agnostic terminology. *nix systems don't usually use the term folder, plus it's a GUI concept anyway (a graphical metaphor for a directory), whereas directories are a file system concept, which is what I think we care about here.

Also remove the Oxford comma since I don't think it's generally used anywhere in the docs.

Reply:
We use the oxford comma all the time, and should prefer it.

Much to my chagrin, we've settled on the term sub-folder here. We use folder instead of directory.

Reply:
Ah, alright. Though I believe the spelling “subfolder” is more common than “sub-folder” (2.26M vs 1.05M Google hits). Wiktionary and the Microsoft Terminology Collection also don't have entries for “sub-folder”, and it's not listed as an alternative spelling for “subfolder” either. So I think the more common spelling should be used.